src/core/textures/Spritesheet.js

```diff
@@ -0,0 +1,269 @@
+
+            if (rect)
+            {
+                let frame = null;
+                let trim = null;
+                const orig = new Rectangle(
+                    0,
+                    0,
+                    this._frames[i].sourceSize.w / this.resolution,
+                    this._frames[i].sourceSize.h / this.resolution
+                );
+
+                if (this._frames[i].rotated)
+                {
+                    frame = new Rectangle(
+                        rect.x / this.resolution,
+                        rect.y / this.resolution,
+                        rect.h / this.resolution,
+                        rect.w / this.resolution
+                    );
+                }
+                else
+                {
+                    frame = new Rectangle(
+                        rect.x / this.resolution,
+                        rect.y / this.resolution,
+                        rect.w / this.resolution,
+                        rect.h / this.resolution
+                    );
+                }
+
+                //  Check to see if the sprite is trimmed
+                if (this._frames[i].trimmed)
+                {
+                    trim = new Rectangle(
+                        this._frames[i].spriteSourceSize.x / this.resolution,
+                        this._frames[i].spriteSourceSize.y / this.resolution,
+                        rect.w / this.resolution,
+                        rect.h / this.resolution
+                    );
+                }
+
+                this.textures[i] = new Texture(
+                    this.baseTexture,
+                    frame,
+                    orig,
+                    trim,
+                    this._frames[i].rotated ? 2 : 0
+                );
+
+                // lets also add the frame to pixi's global cache for fromFrame and fromImage functions
+                TextureCache[i] = this.textures[i];
+            }
+
+            frameIndex++;
+        }
+    }
+
+    /**
+     * The parse has completed.
+     *
+     * @private
+     */
+    _parseComplete()
+    {
+        const callback = this._callback;
+
+        this._callback = null;
+        this._batchIndex = 0;
+        callback.call(this, this.textures);
+    }
+
+    /**
+     * Begin the next batch of textures.
+     *
+     * @private
+     */
+    _nextBatch()
+    {
+        this._processFrames(this._batchIndex * Spritesheet.BATCH_SIZE);
+        this._batchIndex++;
+        setTimeout(() =>
+        {
+            if (this._batchIndex * Spritesheet.BATCH_SIZE < this._frameKeys.length)
+            {
+                this._nextBatch();
+            }
+            else
+            {
+                this._parseComplete();
+            }
+        }, 0);
+    }
+
+    /**
+     * Destroy Spritesheet and don't use after this.
+     */
+    destroy()
+    {
+        for (const i in this.textures)
+        {
+            this.textures[i].destroy();
+            delete this.textures[i];
```

Comment:
Since we are cutting `this.textures` loose instead of retaining it, setting the value to `null` should be faster but just as good as clearing references.